src/renderers/shaders/ShaderChunk/bsdfs.glsl

```diff
@@ -0,0 +1,152 @@
+//#define ENERGY_PRESERVING_MONOCHROME
+
+float calcLightAttenuation( float lightDistance, float cutoffDistance, float decayExponent ) {
+
+	if ( decayExponent > 0.0 ) {
+
+	  return pow( saturate( -lightDistance / cutoffDistance + 1.0 ), decayExponent );
+
+	}
+
+	return 1.0;
+
+}
+
+
+void BRDF_Lambert( const in IncidentLight incidentLight, const in GeometricContext geometryContext, const in vec3 diffuseColor, inout ReflectedLight reflectedLight ) {
+
+	float lambertianReflectance = saturate( dot( geometryContext.normal, incidentLight.direction ) );
+
+	#if defined( ENERGY_PRESERVING_MONOCHROME ) || defined( ENERGY_PRESERVING_RGB )
+
+		lambertianReflectance *= RECIPROCAL_PI;
```

Comment:
@WestLangley:
> @bhouston It is equation 13 for punctual light sources that is implemented in this library. The derivation preceeds the formula in the link.

Did you know "punctual" has not been mentioned in this project in issues or in the source code from my quick search before this particular thread.  I was not aware that we were assuming that all light intensities are pre-multiplied by PI.

This whole discussion is because it was unclear the physical meaning of our light intensities.

I think that for us to be a true physically-based renderer, we need to figure out a unit for our lights.

Frostbite uses "luminous power", the total contribution of each light to the scene, and they specify it in "lumens".  They then convert it to "luminous intensity", the amount of luminous power per area when calculating the light incident on a surface in a particular direction at a particular distance, and they specify in "candelas".